src/app/loginComponents/accounts/accounts.component.ts

```diff
@@ -1,12 +1,53 @@
 import { Component, OnInit } from '@angular/core';
-import { Subject } from 'rxjs';
+import { Location } from '@angular/common';
+import { MatTabChangeEvent } from '@angular/material';
+import { FormControl } from '@angular/forms';
+import { Router } from '@angular/router';
 
 @Component({
   selector: 'app-accounts',
   templateUrl: './accounts.component.html'
 })
 export class AccountsComponent implements OnInit {
+  public currentTab = 'accounts'; // default to the 'accounts' tab
+  protected selected = new FormControl(0);
+  validTabs = ['accounts', 'profiles', 'dockstore account controls', 'requests'];
+  constructor(private location: Location, private router: Router) {}
+
   ngOnInit() {
     localStorage.setItem('page', '/accounts');
+    this.parseURL(this.router.url);
+  }
+
+  private parseURL(url: String): void {
+    const decodedUrl: string = decodeURIComponent(url.toString()); // decode any encoded spaces, etc. in the string
+    const strippedUrl: String = decodedUrl.endsWith('/') ? decodedUrl.slice(0, -1) : decodedUrl; // remove any trailing slash
+    this.setupTab(strippedUrl);
+  }
+
+  public setupTab(url: String) {
+    const regexp: RegExp = new RegExp('\\?tab=(.*)');
```

Comment:
If you keep this (see previous comment), you should move this outside of the method, so that the RegExp is compiled only once.